server/server.go

```diff
@@ -1002,42 +1004,55 @@ func (s *Server) getClientConnectURLs() []string {
 	sPort := strconv.Itoa(opts.Port)
 	urls := make([]string, 0, 1)
 
-	ipAddr, err := net.ResolveIPAddr("ip", opts.Host)
-	// If the host is "any" (0.0.0.0 or ::), get specific IPs from available
-	// interfaces.
-	if err == nil && ipAddr.IP.IsUnspecified() {
-		var ip net.IP
-		ifaces, _ := net.Interfaces()
-		for _, i := range ifaces {
-			addrs, _ := i.Addrs()
-			for _, addr := range addrs {
-				switch v := addr.(type) {
-				case *net.IPNet:
-					ip = v.IP
-				case *net.IPAddr:
-					ip = v.IP
-				}
-				// Skip non global unicast addresses
-				if !ip.IsGlobalUnicast() || ip.IsUnspecified() {
-					ip = nil
-					continue
+	// short circuit if client advertise is set
+	ca := opts.ClientAdvertise
+	if ca != "" {
+		host, port, err := parseHostPort(ca, sPort)
+		s.info.Host = host // TODO: should not set these here.
```

Comment:
This function is called in `routeAcceptLoop`. I thought that you said that you wanted the ClientAdvertise to apply even in standalone, which means changing the server.info.Host/Port so that clients that receive that data don't see the internal host/port. This would not work then.